starter-code/src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health=health;
+    this.strength=strength;
+}
+
+Soldier.prototype.attack=function(){
+    return this.strength;
+}
+Soldier.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name,health,strength) {
+    Soldier.call(this,health,strength);
+    this.name=name;
+    
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+Viking.prototype.battleCry=function(){
+    return "Odin Owns You All!";
+}
 
+Viking.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+    if(this.health-damage>0){
+        return this.name+" has received "+damage +" points of damage";
+    }else {
+        return this.name+" has died in act of combat";
+    }
+}
 // Saxon
-function Saxon() {}
+function Saxon(health,strength) {
+    Soldier.call(this,health,strength);
+}
+
+Saxon.prototype= Object.create(Soldier.prototype);
+Saxon.prototype.receiveDamage=function(damage){
+    this.health-=damage;
+    if(this.health>0){
+        return "A Saxon has received "+damage +" points of damage";
+    }else {
+        return "A Saxon has died in combat";
+    }
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy=[];
+    this.saxonArmy=[];
+}
+function generateViking () {
+    var name     = "Harald";
+    var strength = 150;
+    var health   = 300;
+
+    return new Viking(name, health, strength);
+  }
+
+  function generateSaxon () {
+    var health   = 60;
+    var strength = 25;
+    return new Saxon(health, strength);
+  }
+  War.prototype.addViking = function (viking) {
+    // var viking=new generateViking();
+    this.vikingArmy.push(viking);
+    // return this.vikingArmy;
+  }
+  
+  War.prototype.addSaxon = function (saxon) {
+    this.saxonArmy.push(saxon);
+  }
+      
+  function generateViking() {
+    var name = "Harald";
+    var strength = 150;
+    var health = 300;
+    return new Viking(name, health, strength);
+  }
+  
+  function generateSaxon() {
+    var health = 60;
+    var strength = 25;
+    return new Saxon(health, strength);
+  }
+  
+  War.prototype.vikingAttack=function(){
+    //S- to put out of the function
+      var war = new War()
+      var viking = new generateViking();
+      var saxon = new generateSaxon();
+      war.addViking(viking);
+      war.addSaxon(saxon);
+      // war.addSaxon(saxon);
+      // war.addViking(viking);
+    //E- to put out of the function
+  
+      var v=Math.floor(Math.random()*(war.vikingArmy.length));
+      var s=Math.floor(Math.random()*(war.saxonArmy.length));
+  
+      var myviking=war.vikingArmy[v];
+      var poorsaxon=war.saxonArmy[s];
+      var dmg=myviking.strength;
+      // poorsaxon.receiveDamage(myviking.attack());
+     return poorsaxon.health-dmg;
+  
+  }
```

Comment:
Ok this whole function will not work because you are trying to generate new vikings and saxons inside your function.... what you need to do after you already pushed to the arrays of vikingArmy and saxonArmy  is to generate 2 random numbers that you use as an index on both arrays and they you call the receive damage function of a random one on the other random one depends on who is attacking whom. 

so if a viking is attacking a Saxon then you need call the receive damage function on the RandomSaxon giving it the strength of the RandomViking .. and if the RandomSaxon's health is less than the RandomViking's strength, then you have to remove him from the array as he will be dead. Same thing the other way around.

Reply:
I worked again the methods. Now it will work (I hope :))